File: app/src/main/java/com/king/superdemo/utils/StorageUtil.java
package com.king.superdemo.utils;

import android.content.Context;
import android.os.storage.StorageManager;
import android.os.storage.StorageVolume;
import android.util.Log;

import java.lang.reflect.InvocationTargetException;
import java.lang.reflect.Method;
import java.util.List;

public class StorageUtil {

    public static String getExternalStoragePath(Context context){
        StorageManager storageManager = (StorageManager) context.getSystemService(Context.STORAGE_SERVICE);
        try {
            Method getStorageVolumes = storageManager.getClass().getDeclaredMethod("getStorageVolumes");
            getStorageVolumes.setAccessible(true);
           List<StorageVolume> storageVolumes = (List<StorageVolume>) getStorageVolumes.invoke(storageManager);
            Log.i("wq", "getExternalStoragePath: storageVolumes size="+ storageVolumes.size());
            for (StorageVolume storageVolume : storageVolumes) {
                Log.i("wq", "getExternalStoragePath: "+ storageVolume.getDescription(context) + storageVolume.getUuid());
            }
        } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
            e.printStackTrace();
        }
        return "";
    }

}

File: app/src/main/java/com/king/superdemo/utils/StorageUtil.kt
package com.king.superdemo.utils

import android.content.Context
import android.os.storage.StorageManager
import android.os.storage.StorageVolume
import android.util.Log
import java.lang.reflect.InvocationTargetException

object StorageUtil {
    fun getExternalStoragePath(context: Context): String {
        val storageManager = context.getSystemService(Context.STORAGE_SERVICE) as StorageManager
        try {
            val getStorageVolumes = storageManager.javaClass.getDeclaredMethod("getStorageVolumes")
            getStorageVolumes.isAccessible = true
            val storageVolumes = getStorageVolumes.invoke(storageManager) as List<StorageVolume>
            Log.i("wq", "getExternalStoragePath: storageVolumes size=" + storageVolumes.size)
            for (storageVolume in storageVolumes) {
                Log.i("wq", "getExternalStoragePath: " + storageVolume.getDescription(context) + storageVolume.uuid)
            }
        } catch (e: NoSuchMethodException) {
            e.printStackTrace()
        } catch (e: IllegalAccessException) {
            e.printStackTrace()
        } catch (e: InvocationTargetException) {
            e.printStackTrace()
        }
        return ""
    }
}